crates/sui-network/src/state_sync/mod.rs

```diff
@@ -894,6 +899,60 @@ where
     Ok(())
 }
 
+fn verify_checkpoint<S>(
+    current: &VerifiedCheckpoint,
+    store: S,
+    checkpoint: Checkpoint,
+) -> Result<VerifiedCheckpoint, Checkpoint>
+where
+    S: WriteStore,
+    <S as ReadStore>::Error: std::error::Error,
+{
+    if checkpoint.sequence_number() != current.sequence_number().saturating_add(1)
+        || Some(current.digest()) != checkpoint.previous_digest()
+    {
+        debug!(
+            current_sequence_number = current.sequence_number(),
+            current_digest =% current.digest(),
+            checkpoint_sequence_number = checkpoint.sequence_number(),
+            checkpoint_digest =% checkpoint.digest(),
+            checkpoint_previous_digest =? checkpoint.previous_digest(),
+            "checkpoint not on same chain"
```

Comment:
it seems odd to me to log this message is the sequence number is wrong. i would think this message would be for when the sequence is correct, but the only the digest is wrong.

Reply:
You're correct, it shouldn't be possible. I've change this so that the if condition is clearer and the check for sequence numbers is an assert (just in case there is a bug)